### PR TITLE
BE-1207 Process all branches and save info in commit_detail

### PR DESCRIPTION
**Pull request by xzandrew**

### CODE REVIEW

**Comment by jhaynie** [comment]
Might need to wait until we merge Europa given this is a contract change on commit_activity

**Comment by xzandrew** [comment]
Here is a bit more detail on how it works with incrementals. For example we have a feature branch. On initial run old master commits will be attributed to both master and this feature branch, since it belongs to both.

On incremental run new commits to that branch will be attributed to that branch. But if incremental encounters new feature branch only commits processed in this session will be attributed to that branch (since we don’t process old commits again). If we want to fix this we can write commit details with old hashes and updated branches for old commits (can data import pickup updated fields?).

If some branches were deleted we would lose that info.

File: pkg/git/branches.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	res.BranchedFromCommits = b.BranchedFromCommits
+	res.Commits = b.Commits
+	return
+}
+
+func writeBranches(ctx context.Context, logger log.Logger, rip *ripsrc.Ripsrc, dirs gitDirs) error {
+	started := time.Now()
+	log.Info(logger, "git: generating and writing branches")
+	defer func() {
+		log.Info(logger, "git: done generating and writing branches", "dur", time.Since(started))
```
Comment (jhaynie) [inline pkg/git/branches.go]: let's use our standard `duration` key name we use everywhere else.

File: pkg/git/branches.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	// More stable id that allows name reuse after deletes.
+	ID string `json:"id"`
+
+	// Name of the branch
+	Name string `json:"name"`
+
+	// IsMerged is true if this branch was merged into default branch
+	IsMerged bool `json:"is_merged"`
+
+	// MergeCommit is the hash of the merged commit. Set iff IsMerged=true
```
Comment (jhaynie) [inline pkg/git/branches.go]: `iff` => `if`

**Comment by jhaynie** [comment]
we should merge this into europa once we are ready to merge.

**Comment by xzandrew** [comment]
This currently excludes default branch. I'll add it as well. Processing will have to use that to retrieve master commits only.

**Pull request merged**